Alicerce/Sources/Persistence/DiskMemoryPersistence.swift

```diff
@@ -0,0 +1,326 @@
+    }
+
+    public func removeObject(for key: Persistence.Key, completion: @escaping CompletionClosure<Void>) {
+
+        removeCachedData(for: key)
+
+        removeDiskData(for: key, completion: completion)
+    }
+
+    // MARK: - Private Methods
+
+    // MARK: - NSCache Related Operations
+
+    private func cachedData(for key: Persistence.Key) -> Data? {
+        return cache.object(forKey: key.nsString) as Data?
+    }
+
+    private func setCachedData(_ data: Data, for key: Persistence.Key) {
+        cache.setObject(data.nsData, forKey: key.nsString)
+    }
+
+    private func removeCachedData(for key: Persistence.Key) {
+        cache.removeObject(forKey: key.nsString)
+    }
+
+    // MARK: - Disk Related Operations
+
+    private func diskData(for key: Persistence.Key, completion: @escaping CompletionClosure<Data>) {
+        guard diskCacheEnabled else { return completion { throw PersistenceError.other(Error.diskCacheDisabled) } }
+
+        let readOperation = DiskMemoryBlockOperation() { [unowned self] in
+            let path = self.diskPath(for: key)
+
+            guard let fileData = self.fileManager.contents(atPath: path) else {
+                return completion { throw PersistenceError.other(Error.noDataForKey) }
+            }
+
+            return completion { fileData }
+        }
+
+        readOperationQueue.addOperation(readOperation)
+    }
+
+    private func setDiskData(_ data: Data, for key: Persistence.Key, completion: @escaping CompletionClosure<Void>) {
+        guard diskCacheEnabled else { return completion { throw PersistenceError.other(Error.diskCacheDisabled) } }
+
+        let writeOperation = DiskMemoryBlockOperation() { [unowned self] in
+            let path = self.diskPath(for: key)
+
+            var isDir = ObjCBool(false)
+            let fileExists = self.fileManager.fileExists(atPath: path, isDirectory: &isDir)
+
+            guard isDir.boolValue == false else {
+                return completion { throw PersistenceError.other(Error.fileNotCreated) }
+            }
+
+            guard self.fileManager.createFile(atPath: path, contents: data) else {
+                return completion { throw PersistenceError.other(Error.fileNotCreated) }
+            }
+
+            if !fileExists {
+                self.usedDiskSize += UInt64(data.count)
+            }
+
+            return completion { () }
+        }
+
+        let evictOperation = createEvictOperation()
+        evictOperation.addDependency(writeOperation)
+
+        writeOperationQueue.addOperation(writeOperation)
+        writeOperationQueue.addOperation(evictOperation)
+    }
+
+    private func removeDiskData(for key: Persistence.Key, completion: @escaping CompletionClosure<Void>) {
+        guard diskCacheEnabled else { return completion { throw PersistenceError.other(Error.diskCacheDisabled) } }
+
+        let removeOperation = DiskMemoryBlockOperation(qos: .userInitiated) { [unowned self] in
```

Comment:
The QoS is hardcoded to `.userInitiated`, which is the QoS right after `.userInteractive` (main queue level) 😱. 

Are we sure we need this priority? We could also make it configurable, I guess.